nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,584 @@
+
+
+def colorscale(cmap, values, threshold=None, symmetric_cmap=True):
+    cmap = mpl.cm.get_cmap(cmap)
+    abs_values = np.abs(values)
+    if symmetric_cmap:
+        vmax = abs_values.max()
+        vmin = - vmax
+    else:
+        vmin, vmax = values.min(), values.max()
+    norm = mpl.colors.Normalize(vmin=vmin, vmax=vmax)
+    cmaplist = [cmap(i) for i in range(cmap.N)]
+    abs_threshold = None
+    if threshold is not None:
+        abs_threshold = np.percentile(abs_values, threshold)
+        istart = int(norm(-abs_threshold, clip=True) * (cmap.N - 1))
+        istop = int(norm(abs_threshold, clip=True) * (cmap.N - 1))
+        for i in range(istart, istop):
+            cmaplist[i] = (0.5, 0.5, 0.5, 1.)  # just an average gray color
+    our_cmap = mpl.colors.LinearSegmentedColormap.from_list(
+        'Custom cmap', cmaplist, cmap.N)
+    x = np.linspace(0, 1, 100)
+    rgb = our_cmap(x, bytes=True)[:, :3]
+    rgb = np.array(rgb, dtype=int)
+    colors = []
+    for i, col in zip(x, rgb):
+        colors.append([np.round(i, 3), "rgb({}, {}, {})".format(*col)])
+    return json.dumps(colors), vmin, vmax, our_cmap, norm, abs_threshold
+
+
+def _encode(a):
+    try:
+        data = a.tobytes()
+    except AttributeError:
+        # np < 1.9
+        data = a.tostring()
+    return base64.b64encode(data).decode('utf-8')
+
+
+def _decode(b, dtype):
+    return np.frombuffer(base64.b64decode(b.encode('utf-8')), dtype)
+
+
+def to_plotly(mesh):
+    mesh = surface.load_surf_mesh(mesh)
+    x, y, z = map(_encode, np.asarray(mesh[0].T, dtype='<f4'))
+    i, j, k = map(_encode, np.asarray(mesh[1].T, dtype='<i4'))
+    info = {
+        "_x": x,
+        "_y": y,
+        "_z": z,
+        "_i": i,
+        "_j": j,
+        "_k": k,
+    }
+    return info
+
+
+def _to_color_strings(colors):
+    colors = np.asarray(colors)
+    colors = np.asarray(colors * 255, dtype='uint8')
+    colors = ['#{:02x}{:02x}{:02x}'.format(*row) for row in colors]
+    return colors
+
+
+def _get_vertexcolor(surf_map, cmap, norm,
+                     absolute_threshold=None, sulc_depth_map=None):
+    vertexcolor = cmap(norm(surf_map).data)
+    if absolute_threshold is None:
+        return _to_color_strings(vertexcolor)
+    if sulc_depth_map is None:
+        sulc_depth_map = np.ones(len(surf_map)) * .5
+        anat_vmin, anat_vmax = 0, 1
+    else:
+        sulc_depth_map = surface.load_surf_data(sulc_depth_map)
+        anat_vmin, anat_vmax = np.min(sulc_depth_map), np.max(sulc_depth_map)
+    anat_norm = mpl.colors.Normalize(vmin=anat_vmin, vmax=anat_vmax)
+    anat_color = mpl.cm.get_cmap('Greys')(anat_norm(sulc_depth_map))
+    vertexcolor[np.abs(surf_map) < absolute_threshold] = anat_color[
+        np.abs(surf_map) < absolute_threshold]
+    return _to_color_strings(vertexcolor)
+
+
+def one_mesh_info(surf_map, surf_mesh, threshold=None, cmap=cm.cold_hot,
+                  black_bg=False, bg_map=None, symmetric_cmap=True):
```

Comment:
A small docstring would help here, because I cannot guess what this function does from it's name.